packages/material-ui-lab/src/Pagination/Pagination.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { withStyles } from '@material-ui/core/styles';
+import usePagination from './usePagination';
+import PaginationItem from '../PaginationItem';
+
+export const styles = {
+  /* Styles applied to the root element. */
+  root: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    alignItems: 'center',
+    listStyle: 'none',
+    padding: 0, // Reset
+    margin: 0, // Reset
+  },
+};
+
+const Pagination = React.forwardRef(function Pagination(props, ref) {
+  const {
+    children,
+    classes,
+    className,
+    color = 'standard',
+    getItemAriaLabel: getAriaLabel,
+    items,
+    renderItem = item => <PaginationItem {...item} />,
+    shape = 'round',
+    size,
+    variant = 'text',
+    ...other
+  } = usePagination({ ...props, componentName: 'Pagination' });
+
+  const itemProps = { color, getAriaLabel, shape, size, variant };
+
+  return (
+    <ul
+      role="navigation"
+      aria-label="Pagination navigation"
+      className={clsx(classes.root, className)}
+      ref={ref}
+      {...other}
+    >
+      {children ||
+        items.map(item => (
+          <li key={item.type || item.page.toString()}>{renderItem({ ...item, ...itemProps })}</li>
```

Comment:
```suggestion
          <li key={item.type !== undefined ? item.type : item.page.toString()}>{renderItem({ ...item, ...itemProps })}</li>
```
hidden implicit conversion.